src/main/kubeconfig-manager.ts

```diff
@@ -9,28 +9,15 @@ import logger from "./logger";
 
 export class KubeconfigManager {
   protected configDir = app.getPath("temp");
-  protected tempFile: string;
+  protected tempFile: string = null;
 
-  private constructor(protected cluster: Cluster, protected contextHandler: ContextHandler, protected port: number) { }
+  constructor(protected cluster: Cluster, protected contextHandler: ContextHandler, protected port: number) { }
 
-  static async create(cluster: Cluster, contextHandler: ContextHandler, port: number) {
-    const kcm = new KubeconfigManager(cluster, contextHandler, port);
-
-    await kcm.init();
-
-    return kcm;
-  }
-
-  protected async init() {
-    try {
-      await this.contextHandler.ensurePort();
-      this.tempFile = await this.createProxyKubeconfig();
-    } catch (err) {
-      logger.error(`Failed to created temp config for auth-proxy`, { err });
+  async getPath() {
+    if (!this.tempFile && this.tempFile !== undefined) {
```

Comment:
Doesn't this cover it?
```suggestion
    if (this.tempFile === null) {
```

Reply:
No because we are not being very consistent with the types, on line 40 bellow we assign `this.tempFile = undefined`. Though I don't see why `if (!this.tempFile)` wouldn't be good enough.

Reply:
I'm confused. What you say is true if that was `||` not `&&`. But as written the condition is "when `this.tempFile` is falsey, except if it is `undefined`", right? What exactly are the conditions when `init()` should be called here?

Reply:
When @nevalla  was last touching this file he didn't want to remove that `undefined` since I think he wanted the path not to get recreated after `unlink()`. I don't know why that should be case.

Reply:
@jim-docker you are probably right (with `this.tempFile === null`). I was just following the existing logic here which ended up to be confusing. Basically we want to call init only if tempFile is not set and unlink has not been called (unlink sets this `undefined`).

Reply:
Shouldn't we throw if `this.tempFile === undefined` then? Since that would mean that we would otherwise be returning `undefined`? Where an error would be easier to track down?

Reply:
Refactored.